### DISCUSSION TITLE
lsp-eslint easily trigger "Too many open files" error on macos

### DISCUSSION THREAD

**Discussion by yqrashawn** [emacs-lsp/lsp-mode#3226] [Q&A]
1. lsp client returns `(lsp-session-folders (lsp-session))` as `workspaceFolders` to `lsp-eslint`
1. `lsp-eslint` tell lsp client to watch all the folders
1. which makes emacs process exceeds the default max file descriptors per proc
1. which triggers the "Too many open files" error

I'm not familiar with lsp. So my question might be obvious.

1. is lsp session supposed to be persist?
1. is the above description the supposed way?

**Comment by megane42** [comment]
I completely don't know what happens here, but the following works for me:
https://github.com/emacs-lsp/lsp-mode/issues/1775#issuecomment-643936810

**Comment by benbellick** [comment]
FYI this happens as well with gopls. I'm wondering though, what is the point of having the file watchers on by default? In what context are files being altered without my awareness?

**Reply by benbellick** [reply]
(e.g. maybe you run formatting on your code, but then why not announce that the lsp should reexamine all of the files as a one off?)
